src/Executor/Executor.php

```diff
@@ -222,12 +223,15 @@ private static function getOperationRootType(Schema $schema, OperationDefinition
     private static function executeFieldsSerially(ExecutionContext $exeContext, ObjectType $parentType, $sourceValue, $path, $fields)
     {
         $results = [];
+        $sourceValue = self::completePromiseIfNeeded($sourceValue);
+
         foreach ($fields as $responseName => $fieldASTs) {
             $fieldPath = $path;
             $fieldPath[] = $responseName;
             $result = self::resolveField($exeContext, $parentType, $sourceValue, $fieldASTs, $fieldPath);
 
             if ($result !== self::$UNDEFINED) {
+                $result = self::completePromiseIfNeeded($result);
```

Comment:
Sorry, couldn't find time on week-end, as there is lot of work now, but I remember about it %)